### DISCUSSION TITLE
Building the AWS C++ SDK targeting Windows on a Mac OS machine using mingw

### DISCUSSION THREAD

**Discussion by Jdesign91** [aws/aws-sdk-cpp#3184] [Q&A] [answered]
Hello! I'm curious if it's possible to build the C++ SDK for Windows while on an OSX machine. I've made a few attempts at this using mingw. I've been able to run my command:

cmake .. \
-DTARGET_ARCH=Windows \
-DBUILD_ONLY="core;cognito-idp;s3 \
-DCMAKE_C_COMPILER=x86_64-w64-mingw32-gcc \
-DCMAKE_C_COMPILER=x86_64-w64-mingw32-g++ 

Sucessfully, but then I fail to run my

make install

command, generally due to things like:

Warnings about using NULL in arithmetic
Unknown pragmas (MSVC specific ones)
Warnings being treated as errors
Unused but set variables.

Any support or direction is appreciated!

**Comment by jmklix** [comment, marked as answer]
Yes, it is possible but it can be rather complicated. We don't have documentation on how to specifically cross compiling this sdk as it is very dependent on your source and target OS. Here are a few links that go into more detail about this. Please let me know if you have any specific questions about cross compiling. I will try to help, but I don't have much experience with cross compiling.
- https://stackoverflow.com/questions/25153044/c-cross-platform-compiling
- https://cmake.org/cmake/help/book/mastering-cmake/chapter/Cross%20Compiling%20With%20CMake.html
- https://github.com/AllenDang/giu/issues/101
